ui/utils_test.py

```diff
@@ -0,0 +1,54 @@
+"""Tests for utils methods"""
+import os
+
+import pytest
+
+from ui import utils
+
+
+# pylint: disable=unused-argument
+
+
+def test_get_moira_client_nox509(mocker):
+    """Test that the correct error message is returned if no x509 certificate file exists"""
+    os.remove(utils.MIT_WS_CERTIFICATE_FILE)
+    os.remove(utils.MIT_WS_PRIVATE_KEY_FILE)
+    mock_moira = mocker.patch('ui.utils.Moira')
+    with pytest.raises(RuntimeError) as err:
+        utils.get_moira_client()
+        assert utils.MIT_WS_CERTIFICATE_FILE in str(err)
+        assert utils.MIT_WS_PRIVATE_KEY_FILE in str(err)
+    assert not mock_moira.called
+
+
+def test_get_moira_client_nokey(mocker):
+    """Test that the correct error message is returned if no x509 key file exists"""
+    utils.write_x509_files()
+    os.remove(utils.MIT_WS_PRIVATE_KEY_FILE)
+    mock_moira = mocker.patch('ui.utils.Moira')
+    with pytest.raises(RuntimeError) as err:
+        utils.get_moira_client()
+        assert utils.MIT_WS_PRIVATE_KEY_FILE in str(err)
+        assert utils.MIT_WS_CERTIFICATE_FILE not in str(err)
+    assert not mock_moira.called
+
+
+def test_get_moira_client_nocert(mocker):
+    """Test that the correct error message is returned if no x509 certificate & key files exist"""
+    utils.write_x509_files()
+    os.remove(utils.MIT_WS_CERTIFICATE_FILE)
+    mock_moira = mocker.patch('ui.utils.Moira')
+    with pytest.raises(RuntimeError) as err:
+        utils.get_moira_client()
+        assert utils.MIT_WS_CERTIFICATE_FILE in str(err)
+        assert utils.MIT_WS_PRIVATE_KEY_FILE not in str(err)
+    assert not mock_moira.called
```

Comment:
I think we should change the approach here.

1. If we can't avoid involving the file system for these tests, we should specify new file paths so that files that exist in the default path aren't altered during the test run. 
1. We should just create empty files at those paths as necessary instead of invoking `write_x509_files`. `get_moira_client` just depends on those files existing. Invoking `write_x509_files` to satisfy that requirement isn't as explicit about what `get_moira_client` expects, and it also unnecessarily extends the scope of this test
1. We can/probably should parametrize the tests where one or both files don't exist. All 3 of them specify some file that shouldn't exist, then check that the missing file path appears in the exception message. I don't think it's necessary to check for an existing file path to be absent from the error message
1. `mock_moira` is created in every test case. We should create a fixture for that patch